appinventor/appengine/src/com/google/appinventor/client/DesignToolbar.java

```diff
@@ -405,11 +408,20 @@ private boolean switchToProject(long projectId, String projectName) {
         OdeLog.wlog("DesignToolbar: ignoring call to switchToProject for current project");
         return true;
       }
-      pushedScreens.clear();    // Effectively switching applications clear stack of screens
+
+      // Clear screens.
+      pushedScreens.clear();  // Effectively switching applications; clear stack of screens.
       clearDropDownMenu(WIDGET_NAME_SCREENS_DROPDOWN);
+      if (currentProject != null) {
+        for (Screen s : currentProject.screens.values()) {
```

Comment:
I'm not sure why this is necessary. The screens haven't actually been removed from the project, so to remove them seems odd to me.

Reply:
Yeah this was actually because of the .prototype sharing you mentioned! No clue how I've been coding js for this long and didn't know about that haha. Since I removed that and switched to the ProjectChangeListener strategy I removed this as well :D